scale/src/commonMain/kotlin/Ticks.kt

```diff
@@ -0,0 +1,89 @@
+package com.juul.krayon.scale
+
+import kotlin.math.abs
+import kotlin.math.floor
+import kotlin.math.log10
+import kotlin.math.pow
+import kotlin.math.roundToInt
+import kotlin.math.sqrt
+
+// A friendly reminder to thank your local math-magician, because I don't know what's happening in most of this file.
+
+private val sqrt50 = sqrt(50.0)
+private val sqrt10 = sqrt(10.0)
+private val sqrt2 = sqrt(2.0)
+
+/** See equivalent [d3 function](https://github.com/d3/d3-array#tick). */
+public fun ticks(start: Float, stop: Float, count: Int): List<Float> =
+    ticks(start.toDouble(), stop.toDouble(), count).map { it.toFloat() }
+
+/** See equivalent [d3 function](https://github.com/d3/d3-array#tick). */
+public fun ticks(start: Double, stop: Double, count: Int): List<Double> {
```

Comment:
any idea why there's so much logic in here for it to be reversible? seems like a strange priority unless im missing something

Reply:
The logic for reversing is just a couple lines, to make sure that it plays nice with `tickIncrement`.

The larger calculation blocks are actually handling if ticks increment by a positive or negative power of `10`. I don't understand the math well enough to know why the math has to be different for those, but 🤷